### ISSUE TITLE
Build fails for legal PEP-440 version identifiers that are not x.y.z

### ISSUE THREAD

**Issue by tyteen4a03** [panda3d/panda3d#1539]
## Description
Panda3D's build system does not seem to support non-`x.y.z`-formatted [PEP-440](https://peps.python.org/pep-0440/) versions. 

Examples of errors include: 
```
# 1.11.0a1
 Traceback (most recent call last):
  File "/home/runner/actions-runner/_work/panda3d/panda3d/makepanda/makepanda.py", line 2753, in <module>
    CreatePandaVersionFiles()
  File "/home/runner/actions-runner/_work/panda3d/panda3d/makepanda/makepanda.py", line 2704, in CreatePandaVersionFiles
    version3=int(VERSION.split(".")[2])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '0a1'
```
```
# 1.11.0.dev1
ld: malformed 64-bit a.b.c.d.e version number: 1.11.0.dev1
```

## Steps to Reproduce
1. Change the version number of Panda3D in setup.cfg to 1.11.0.dev1 or 1.11.0a1.
2. Build Panda3D.

## Environment
 * Operating system: macOS
 * System architecture: x64
 * Panda3D version: N/A
 * Installation method: Built from source
 * Python version (if using Python): 3.11
 * Compiler (if using C++): clang

**Comment by rdb** [comment]
Addressed in dcc96a60b1a7a6259c1e8f50595b242359fb8501, will merge to master soon

**Comment by rdb** [comment]
Actually, it should strip the suffixes for `VERSION` but only keep them for `WHLVERSION`, that is the same behaviour as with `--version`
